### PR TITLE
ConfigurationGathering.py: Change COAFILE_OUTPUT to refer to .coafile

### PR DESCRIPTION
**Pull request by PrajwalM2212**
This changes COAFILE_OUTPUT to refer to .coafile

Closes https://github.com/coala/coala/issues/5417

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
@jayvdb  `COAFILE_OUTPUT` is used only in the function `load_config_file`. The message gives instructions to create one as `--save` creates a `.coafile` .
screen shot 2018-05-02 at 6 46 12 pm

But the function `load_config_file` is called from `load_configuration` function . 
screen shot 2018-05-02 at 7 41 12 pm
As you said `load_config_file` is ironically called only for loading `User-Wide and System-Wide coafile`. 

So now you wanted me to change the message of `COAFILE_OUTPUT` to show how to create a `User-Wide or System-Wide coafile` instead of a `.coafile` right ?

**Comment by PrajwalM2212** [comment]
@jayvdb  It has 2 invocations . One is for` system/default coafile` and other is for `user coafile`. I have changed the else part to accommodate for `user/.coarc coafile` and left the if part as it is .

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -16,6 +16,13 @@
 from coalib.settings.Setting import Setting, path
 from string import Template
 
+COARC_OUTPUT = Template('$type \'$file\' $found!\n'
+                        'Here\'s what you can do:\n'
+                        '*add `--config .coarc --save` '
```
Comment (jayvdb) [inline coalib/settings/ConfigurationGathering.py]: this will be used when loading the `get_system_coafile` if it returns the file named `default_coafile`

Comment (jayvdb) [reply coalib/settings/ConfigurationGathering.py]: Also where is the `.coarc` located?

**Comment by jayvdb** [comment]
ping @PrajwalM2212 .  https://github.com/coala/coala/issues/5440 is ready to roll.  Could be a simple extra commit in this PR.

**Comment by PrajwalM2212** [comment]
@jayvdb I have sent PR for #5440 . I will update this after that is merged,

**Comment by jayvdb** [comment]
Test case missing ... coverage is complaining missing coverage for one branch and one line.

**Comment by jayvdb** [comment]
Focus on ConfigurationGatheringTest.py ; in there should be a simple test which you can borrow.

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**